stdlib/Test/src/Test.jl

```diff
@@ -1932,36 +1932,43 @@ function testset_beginend_call(args, tests, source)
             $(testsettype)($desc; $options...)
         end
 
-        # we reproduce the logic of guardseed, but this function
-        # cannot be used as it changes slightly the semantic of @testset,
-        # by wrapping the body in a function
-        local default_rng_orig = copy(default_rng())
-        local tls_seed_orig = copy(Random.get_tls_seed())
-        local ts_rng = get_rng(ts)
-        local tls_seed = isnothing(ts_rng) ? set_rng!(ts, tls_seed_orig) : ts_rng
-        try
-            @with_testset ts begin
-                # default RNG is reset to its state from last `seed!()` to ease reproduce a failed test
-                copy!(Random.default_rng(), tls_seed)
-                copy!(Random.get_tls_seed(), Random.default_rng())
-                let
-                    $(esc(tests))
+        # Check if this testset should run based on filtering
+        local testset_id = generate_deterministic_testset_id($(string(source.file)), $(source.line), $desc)
```

Comment:
Does this mean that adding some newlines changes this ID? Or adding new lines to a macro wrapping `@testset`?

Reply:
The ID just needs to be stable between the discovery and runtime. It doesn't need to be stable otherwise.